Core/PersistentPixel.swift

```diff
@@ -299,22 +301,13 @@ public final class PersistentPixel: PersistentPixelFiring {
 
         Logger.general.debug("Saving persistent pixel named \(pixelMetadata.pixelName)")
 
-        if self.isSendingQueuedPixels {
-            self.failedPixelsPendingStorage.append(pixelMetadata)
-        } else {
-            try self.persistentPixelStorage.append(pixel: pixelMetadata)
-        }
+        try self.persistentPixelStorage.append(pixels: [pixelMetadata])
```

Comment:
It's now possible to write pixels to storage even when the queue is actively being processed, since we now only remove pixels based on ID - newly failed pixels will be untouched at the end of processing.